src/lib/libwasmfs_fetch.js

```diff
@@ -35,34 +31,74 @@ addToLibrary({
         try {
           var u = new URL(fileUrl, self.location.origin);
           url = u.toString();
-        } catch (e) {
+        } catch (_e) {
+          throw {status: 404};
         }
       }
-      var response = await fetch(url);
-      if (response.ok) {
-        var buffer = await response['arrayBuffer']();
-        wasmFS$JSMemoryFiles[file] = new Uint8Array(buffer);
-      } else {
+      var chunkSize = __wasmfs_fetch_get_chunk_size(file);
+      offset ??= 0;
+      len = len || chunkSize;
+      var firstChunk = (offset / chunkSize) | 0;
+      var lastChunk = ((offset+len) / chunkSize) | 0;
+      if (!(file in wasmFS$JSMemoryRanges)) {
+        var fileInfo = await fetch(url,{method:"HEAD", headers:{"Range": "bytes=0-"}});
+        if (fileInfo.ok &&
+           fileInfo.headers.has("Content-Length") &&
+           fileInfo.headers.get("Accept-Ranges") == "bytes" &&
+           (parseInt(fileInfo.headers.get("Content-Length")) > chunkSize*2)) {
+          wasmFS$JSMemoryRanges[file] = {size:parseInt(fileInfo.headers.get("Content-Length")), chunks:[], chunkSize:chunkSize};
```

Comment:
```suggestion
          wasmFS$JSMemoryRanges[file] = {
            size: parseInt(fileInfo.headers.get("Content-Length")),
            chunks: [],
            chunkSize: chunkSize
          };
```
this matches our general formatting a little better